### PR TITLE
perf: Booker-Bundle Size Improvement - Lazy load react-select, remove lodash and Dom Purify

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
- Removed lodash usage from Booker Pages
	- Using lodash now throws eslint error. All existing usages whitelisted that were not on Booker pages. 
- Removed DOM Purify usage from all pages
- Lazy loading react-select as it's not needed by System Fields and can be loaded when needed.

Reduces  payload for Booker Pages by ~12%. Also reduces the payload for other pages by significant numbers

Screenshot 2023-08-18 at 3 54 27 PM

Fixes # (issue)

## Type of change
- [x] Perf Improvement

## How should this be tested?
- [ ] E2E Booking creation with a Select type question
- [ ] Managed Event Type Locked/Unlocked fields testing
- [ ] Verify SEO tags

## Mandatory Tasks
- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/ui/components/form/inputs/Input.tsx [hunk 1]
```diff
@@ -39,174 +18,6 @@ export function InputLeading(props: JSX.IntrinsicElements["div"]) {
...
-      className={classNames(
-        "min-h-9 flex flex-col justify-center text-sm leading-7",
-        error ? "text-error" : "text-default"
-      )}>
-      <span className="flex whitespace-nowrap">{children}</span>
-    </div>
-  </div>
-);
-
-export const InputField = forwardRef<HTMLInputElement, InputFieldProps>(function InputField(props, ref) {
```
Comment (hariombalhara) [inline packages/ui/components/form/inputs/Input.tsx:86]: Webpack not able to do tree shaking for this module very well, so move the components in separate files.

File: packages/ui/components/form/inputs/InputFieldWithSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, { forwardRef } from "react";
+
+import { InputField, UnstyledSelect } from "../../..";
+import type { InputFieldProps } from "./types";
+
+export const InputFieldWithSelect = forwardRef<
```
Comment (hariombalhara) [inline packages/ui/components/form/inputs/InputFieldWithSelect.tsx:6]: Keeping this in separate module from` Input.tsx` as it imports select

File: packages/app-store/routing-forms/components/react-awesome-query-builder/widgets.tsx [hunk 3]
```diff
@@ -7,9 +8,13 @@ import type {
   ProviderProps,
 } from "react-awesome-query-builder";
 
-import { Button as CalButton, SelectWithValidation as Select, TextField, TextArea } from "@calcom/ui";
+import { Button as CalButton, TextField, TextArea } from "@calcom/ui";
 import { Trash, Plus } from "@calcom/ui/components/icon";
 
+const Select = dynamic(
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/components/react-awesome-query-builder/widgets.tsx:14]: `Select` is not used by any system field thus it doesn't make sense to load it as part of initial bundle.

File: packages/features/ee/common/components/LicenseRequired.tsx [hunk 4]
```diff
@@ -63,17 +62,17 @@ const LicenseRequired = ({ children, as = "", ...rest }: LicenseRequiredProps) =
           headline={t("enterprise_license")}
           description={
             <div
+              // It is okay to not sanitize the HTML here, because the variables in here are not user-provided. They are env variables hardcoded in the instance.
+              // In turn it saves us the cost of super heavy dom purify library
               dangerouslySetInnerHTML={{
```
Comment (hariombalhara) [inline packages/features/ee/common/components/LicenseRequired.tsx:67]: Translated strings can be compromised here. So, it's not safe.

File: packages/features/ee/common/components/LicenseRequired.tsx [hunk 5]
```diff
@@ -62,20 +61,17 @@ const LicenseRequired = ({ children, as = "", ...rest }: LicenseRequiredProps) =
...
-                ${APP_NAME}
-              </a>`,
-                    setupUrl: `<a href="${WEBAPP_URL}/auth/setup" class="underline">/auth/setup</a>`,
-                    supportMail: `<a href="mailto:${SUPPORT_MAIL_ADDRESS}" class="underline">
-                ${SUPPORT_MAIL_ADDRESS}</a>`,
-                  })
-                ),
-              }}
-            />
+            <Trans i18nKey="enterprise_license_description">
```
Comment (hariombalhara) [inline packages/features/ee/common/components/LicenseRequired.tsx:64]: Verified that the translation is being picked up 
image

**Review by zomars — APPROVED** [review]
Noice

**Pull request merged**